### PR TITLE
Smart commit

### PR DESCRIPTION
**Pull request by jreidinger**

### CODE REVIEW

**Comment by schubi2** [comment]
LGTM

File: lib/tasks/osc.rake [hunk 1]
```diff
@@ -81,6 +82,41 @@ namespace :osc do
...
+    Dir.mktmpdir("unpacked_tarball") do |d|
+      sh "tar xvf #{source1} -C #{d}"
+
+      Dir.mktmpdir("unpacked_tarball") do |d2|
+        sh "tar xvf #{source2} -C #{d2}"
+
+        res = `diff -r #{d} #{d2}`
+        puts res if verbose
+
+        return $?.exitstatus != 0
```
Comment (lslezak) [inline lib/tasks/osc.rake]: You can use `return $?.success?` instead here.

File: lib/tasks/osc.rake [hunk 2]
```diff
@@ -81,6 +82,41 @@ namespace :osc do
...
+    Dir["#{package_dir}/*"].each do |f|
+      orig = f.sub(/#{package_dir}\//, "")
+      if orig =~ /\.(tar|tbz2|tgz|tlz)/ # tar archive, so ignore archive creation time otherwise it always looks like new one
+        return if different_tarballs?(f, File.join(osc_checkout_dir, orig))
+      else
+        cmd = "diff #{f} #{osc_checkout_dir}/#{orig}"
+        puts cmd if verbose
+        puts `bash -c '#{cmd}'`
+
+        return if $?.exitstatus != 0 # there is something new
```
Comment (lslezak) [inline lib/tasks/osc.rake]: Similarly `return !$?.success?` here.

File: lib/tasks/osc.rake [hunk 3]
```diff
@@ -81,6 +82,41 @@ namespace :osc do
     version
   end
 
+  def different_tarballs?(source1, source2)
+    Dir.mktmpdir("unpacked_tarball") do |d|
```
Comment (lslezak) [inline lib/tasks/osc.rake]: I'd prefer `d1` and `d2` (to be more consistent).

File: lib/tasks/osc.rake [hunk 4]
```diff
@@ -81,6 +82,41 @@ namespace :osc do
...
+      sh "tar xvf #{source1} -C #{d}"
+
+      Dir.mktmpdir("unpacked_tarball") do |d2|
+        sh "tar xvf #{source2} -C #{d2}"
+
+        res = `diff -r #{d} #{d2}`
```
Comment (lslezak) [inline lib/tasks/osc.rake]: I'd prefer unified diff using `-u` option.

File: package/rubygem-packaging_rake_tasks.changes [hunk 5]
```diff
@@ -1,4 +1,10 @@
 -------------------------------------------------------------------
+Tue Apr 28 15:51:40 UTC 2015 - jreidinger@suse.com
+
+- stop osc commit if there is no changes
```
Comment (lslezak) [inline package/rubygem-packaging_rake_tasks.changes]: Either `is no change` or `are no changes`.

**Comment by lslezak** [comment]
LGTM, thanks ! :+1:

**Pull request merged**